### PR TITLE
Add structured logging to logs input

### PR DESCRIPTION
**Pull request by urso**
- Enhancement

## What does this PR do?

The logging of the logs input is updated to make use of structured
logging. Each logs input instance will create an unique ID, that will be
logged as "input_id".
The input and harvester will also add file state information like:
source, state_id, finished, os_id. The os_id holds the inode, and
state_id is the ID that is used internally by filebeat to track the
file.

## Why is it important?

By filtering the logs for one or the other ID, one can track all
logs at a different level. Filtering by "state_id" allows us to monitor
all logs and state changes for a given file, even if the file is
renamed, closed and reopened.

## Checklist

- [x] My code follows the style guidelines of this project
~~- [ ] I have commented my code, particularly in hard-to-understand areas~~
~~- [ ] I have made corresponding changes to the documentation~~
~~- [ ] I have made corresponding change to the default configuration files~~
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

Run filebeat with the logs input and debug logs enabled.

File state updates after a scan or harvester logs should log contextual information like the inode, state_id, harvester_id, input_id.

### CODE REVIEW

File: filebeat/input/log/harvester.go [hunk 1]
```diff
@@ -318,21 +325,21 @@ func (h *Harvester) Run() error {
 		if err != nil {
 			switch err {
 			case ErrFileTruncate:
-				logp.Info("File was truncated. Begin reading file from offset 0: %s", h.state.Source)
+				logger.Infof("File was truncated. Begin reading file from offset 0: %s", h.state.Source)
```
Comment (kvch) [inline filebeat/input/log/harvester.go:328]: Nit: as the path is now part of the log, there is no need for adding `h.state.Source` to the log line.

**Review by kvch — APPROVED** [review]

File: filebeat/input/log/harvester.go (hunk 1 above)
Comment (urso) [reply filebeat/input/log/harvester.go:328]: True, I'm kinda lazy here, as I didn't want to break all integration tests (some are already broken due to changes in the log output).

Plus in Logs UI or Kibana discover, one might still look for the full message in the first place. Here the additional fields mostly provide filtering capabilities.

**Review by kvch — APPROVED** [review]

**Comment by ph** [comment]
@urso is this something you want  to add for 7.14?

**Comment by urso** [comment]
> is this something you want to add for 7.14?

The item is 'complete' with this PR, so yeah it is 7.14 :)
As this helps debugging potential issues (correlating logs), this change goes under the unbrella of supportability.

**Pull request merged**